src/solvers/refinement/string_refinement.cpp

```diff
@@ -1137,6 +1137,33 @@ static exprt negation_of_constraint(const string_constraintt &axiom)
   return negaxiom;
 }
 
+/// Result of the solver `supert` should not be interpreted literaly for char
+/// arrays as not all indices are present in the index set.
+/// We populate the values for which the solver has not constraint by padding
```

Comment:
Give a concrete example of the sort of change this produces (e.g. array_of(0) with 1 := 2 -> { 0, 2, 0} ?)